File: src/test/java/jodajs/TestUtils.java
package jodajs;

import org.joda.time.LocalDate;
import org.joda.time.LocalDateTime;
import org.joda.time.LocalTime;
import org.joda.time.chrono.GJChronology;
import org.joda.time.format.DateTimeFormatterBuilder;

import java.util.Date;
import java.util.Locale;

/**
 *
 */
public class TestUtils {
    static {
        Locale.setDefault(Locale.ENGLISH);
    }

    public static LocalDateTime LocalDateTime(int y, int m, int d, int h, int mi, int s, int ms) {
        return new LocalDateTime(y, m, d, h, mi, s, ms);
    }

    public static LocalDateTime LocalDateTime(int y, int m, int d, int h, int mi, int s) {
        return new LocalDateTime(y, m, d, h, mi, s);
    }

    public static LocalDateTime LocalDateTime(int y, int m, int d, int h, int mi) {
        return new LocalDateTime(y, m, d, h, mi);
    }

    public static LocalTime LocalTime(int h, int mi, int s, int ms) {
        return new LocalTime(h, mi, s, ms);
    }

    public static LocalTime LocalTime(int h, int mi) {
        return new LocalTime(h, mi);
    }

    public static LocalDate LocalDate(int y, int m, int d) {
        return new LocalDate(y, m, d, GJChronology.getInstance());
    }

    public static DateTimeFormatterBuilder DateTimeFormatterBuilder() {
        return new DateTimeFormatterBuilder();
    }

    public static Date stdDate(int y, int m, int d, int h, int mi, int s, int ms) {
        return new Date(new Date(y - 1900, m, d, h, mi, s).getTime() + ms);
    }

}

File: src/test/js/TestUtils.js
/*global beforeEach,jasmine,jodajs*/
var timeZone = new Date().getTimezoneOffset() * 60 * 1000,

    littleBefore = function (time) {
        return time - 10;
    },
    almostDayAfter = function (time) {
        return time + 24 * 60 * 60 * 1000 - 10;
    },
    goodDayAfter = function (time) {
        return time + 24 * 60 * 60 * 1000 + 10;
    },

    stdDate = function (y, m, d, h, mi, s, ms) {
        return new Date(y, m, d, h, mi, s, ms);
    },

    DateTimeUtils = jodajs.DateTimeUtils,
    DateTimeFormatterBuilder = jodajs.DateTimeFormatterBuilder,
    DateTimeFormat = jodajs.DateTimeFormat,
    JsonFormatter = jodajs.JsonFormatter,
    LocalDateTime = jodajs.LocalDateTime,
    LocalTime = jodajs.LocalTime,
    LocalDate = jodajs.LocalDate;

beforeEach(function () {
    this.addMatchers({
        toEq: function (expected) {
            if (typeof this.actual.isEqual !== 'function') {
                this.message = function (expected) {
                    return this.actual + ' does not have an isEqual function';
                };
                return false;
            }
            if (!this.actual.isEqual(expected)) {
                this.message = function (expected) {
                    return this.actual.toDate().toUTCString() + ' is expected to be equal ' + expected.toDate().toUTCString();
                };
                return false;
            }
            return true;
        },
        toMatch: function (expected) {
            var regex = expected.test ? expected : new RegExp('^' + expected + '$');
            if (!this.actual.match(regex)) {
                return false;
            }
            return true;
        }
    });
});